plugins/destination/plugin.go

```diff
@@ -3,54 +3,127 @@ package destination
 import (
 	"context"
 	"fmt"
+	"sync"
 	"time"
 
 	"github.com/cloudquery/plugin-sdk/schema"
 	"github.com/cloudquery/plugin-sdk/specs"
 	"github.com/rs/zerolog"
-	"golang.org/x/sync/errgroup"
+)
+
+type writerType int
+
+const (
+	unmanaged writerType = iota
+	managed
 )
 
 type NewClientFunc func(context.Context, zerolog.Logger, specs.Destination) (Client, error)
 
+type ManagedWriter interface {
+	PreWriteTableBatch(ctx context.Context, tables schema.Tables) error
+	WriteTableBatch(ctx context.Context, table *schema.Table, data [][]interface{}) error
+}
+
+type UnimplementedManagedWriter struct{}
+
+type UnmanagedWriter interface {
+	Write(ctx context.Context, tables schema.Tables, res <-chan *ClientResource) error
+	Metrics() Metrics
+}
+
+type UnimplementedUnmanagedWriter struct{}
+
+func (*UnimplementedManagedWriter) PreWriteTableBatch(context.Context, schema.Tables) error {
+	panic("PreWriteTableBatch not implemented")
+}
+
+func (*UnimplementedManagedWriter) WriteTableBatch(context.Context, *schema.Table, [][]interface{}) error {
+	panic("WriteTableBatch not implemented")
+}
+
+func (*UnimplementedUnmanagedWriter) Write(context.Context, schema.Tables, <-chan *ClientResource) error {
+	panic("Write not implemented")
+}
+
+func (*UnimplementedUnmanagedWriter) Metrics() Metrics {
+	panic("Metrics not implemented")
+}
+
 type Client interface {
 	schema.CQTypeTransformer
-	ReverseTransformValues(table *schema.Table, values []any) (schema.CQTypes, error)
+	ReverseTransformValues(table *schema.Table, values []interface{}) (schema.CQTypes, error)
 	Migrate(ctx context.Context, tables schema.Tables) error
-	Read(ctx context.Context, table *schema.Table, sourceName string, res chan<- []any) error
-	Write(ctx context.Context, tables schema.Tables, res <-chan *ClientResource) error
-	Metrics() Metrics
+	Read(ctx context.Context, table *schema.Table, sourceName string, res chan<- []interface{}) error
+	ManagedWriter
+	UnmanagedWriter
 	DeleteStale(ctx context.Context, tables schema.Tables, sourceName string, syncTime time.Time) error
 	Close(ctx context.Context) error
 }
 
 type ClientResource struct {
 	TableName string
-	Data      []any
+	Data      []interface{}
 }
 
+type Option func(*Plugin)
+
 type Plugin struct {
 	// Name of destination plugin i.e postgresql,snowflake
 	name string
 	// Version of the destination plugin
 	version string
 	// Called upon configure call to validate and init configuration
-	newDestinationClient NewClientFunc
+	newClient NewClientFunc
+	writer    writerType
```

Comment:
```suggestion
	writerType    writerType
```

Let's call this field `writerType` so it's not confused with an actual writer